protocol.md

```diff
@@ -726,6 +727,120 @@ Upload-Length: 11
 Upload-Concat: final;/files/a /files/b
 ```
 
+### Client Tag
+
+With this extension, Clients can provide a Tag for an upload - with which they
+can later retrieve the URL for the created resource.
+
+This allows Clients to resume an upload initiated through the [Creation With Upload](#creation-with-upload)
+or [Creation](#creation) extensions for which they did not receive the response with
+the newly created resource's URL (for example due to a broken connection).
+
+If the Server supports this extension, it MUST add `client-tag` to the `Tus-Extension`
+header.
+
+To initiate an upload with Client Tag, Clients MUST include an `Upload-Tag`
+header with the initial `POST` request (as specified in the [Creation](#creation) and
+[Creation With Upload](#creation-with-upload) extensions) made to the Upload URL.
+
+Clients can then send a `HEAD` request with the same `Upload-Tag` header to the
+Upload URL, which - upon success - MUST respond with the `200 OK` or `204 No Content`
+status, and the resource's URL in the response's `Location` header.
+
+A successful response MUST also contain the `Tus-Version` header. An `Upload-Offset` header
+that indicates how many bytes have already been received by the server MAY also be returned.
+
+Clients can then use the returned `Location` to resume the upload.
+
+If an `Upload-Tag` is unknown or no longer known by the Server, it MUST
+respond to `HEAD` requests with the `404 Not Found` or `410 Gone` status.
+
+If an `Upload-Tag` is already in use for a different, ongoing upload, the Server
+MUST respond to the `POST` request initiating the upload with a `409 Conflict` status.
+
+#### Security Considerations
+
+To protect against attackers trying to guess an `Upload-Tag` and subsequently
+use it to inject malicious content into an upload, Servers MUST take measures to
+ensure that only the party that created a resource with an `Upload-Tag` can use it
+for subseqeuent `PATCH` and `HEAD` requests.
+
+If uploading to a Server is only possible for authenticated users, the Server can
+satisfy this requirement by leveraging the available authentication information to
+bind the `Upload-Tag` to a particular user, so that only that user can use it.
+
+Servers MUST return a `403 Forbidden` status for unauthenticated requests that
+contain an `Upload-Tag` header.
+
+#### Headers
+
+##### Upload-Tag
+
+The `Upload-Tag` request header MUST be an identifier created by the
+Client that's unique within its storage space, to avoid collisions.
+
+Clients can satisfy this requirement by generating and using a [version 4 UUID](https://tools.ietf.org/html/rfc4122#section-4.4).
+
+A `Upload-Tag` MUST consist of ASCII characters and MUST NOT exceed 200 bytes.
```

Comment:
What's the reason for the 200 bytes limit?

Reply:
I argued for some limit on the upload tag, so server can rely on them. However, there is not specific reason for the number 200. Open for other suggestions :)

Reply:
I don't really have any argument here, just wondering if there was a specific reason :) 200 bytes is fine by me.

Reply:
We could also make it 256 or similar to allow SHA-256 to be used.

Reply:
I don't have a strong opinion there either. I'd be fine with both 200 and 256. :)

Reply:
Let's go with 256 characters. When doing hexadecimal encoding, one can fit 1024 bits into the Upload-Tag which is enough to hold nearly all checksums :) I can update the PR